### PR TITLE
fix #7753 type(chore): Update targeting integration tests to use graphQL API

### LINKED ISSUE #6791: Integration test failure: targeting tests being tricked by targeting slug
Just saw the weirdest behaviour in the targeting integration tests.  We pass a couple parameters into the test, one is an application slug like `DESKTOP`, and the other is a targeting config slug.  If the targeting config slug also includes the word `desktop` like `something_desktop_something` then it will trick pytest into running the tests with applications other than desktop which will then fail.

https://app.circleci.com/pipelines/github/mozilla/experimenter/11753/workflows/223c1fac-3cd5-4d94-a25b-6c07d0bbd489/jobs/35944

We've probably got to dig into a better way to parameterize the tests so it doesn't get tricked into running the targeting tests on applications other than desktop.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/EXP-1940)

### LINKED ISSUE #7753: Update targeting integration tests to use graphql API
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2801)

### PR DESCRIPTION
**Pull request by jrbenny35**
Because:
* Using the UI to create experiments for the targeting tests can take a lot of time

This Commit:
* Switches to using the graphQL API to create experiments resulting in faster tests.

Also fixes #6791

### CODE REVIEW

File: app/tests/integration/nimbus/conftest.py [hunk 1]
```diff
@@ -320,6 +313,48 @@ def _language_database_id_loader(languages=None):
...
+        path = os.path.join("/", path, "experimenter/base/fixtures/locales.json")
+        with open(path) as file:
+            data = json.loads(file.read())
+            for locale in locales:
+                for item in data:
+                    if locale in item["fields"]["code"]:
+                        locale_list.append(item["pk"])
+        return locale_list
+
+    return _locales_database_id_loader
```
Comment (jrbenny35) [inline app/tests/integration/nimbus/conftest.py:355]: I could probably write one function to do all of this, but maybe I save that for a contributor??

Comment (jaredlockhart) [reply app/tests/integration/nimbus/conftest.py:355]: Actually the other thing I was thinking is you could also query this info from the configuration endpoint.  I thought we were doing that at some point but now I don't see it.  Or did we figure out that it was too slow/janky?  Reading them from the disk works too.

Comment (jrbenny35) [reply app/tests/integration/nimbus/conftest.py:355]: I'd need help with that if you think it's worth doing.

**Review by jaredlockhart — APPROVED** [review]
Yep all looks good, thank you, nice optimization.  I think we should keep this pattern up and use the api for anything that isn't explicitly testing the nimbus ui.  Thanks @jrbenny35 !

Oh also I just noticed, the commit formatting is slightly off, it's not

```
type(chore)
```

it should be

```
chore(nimbus)
```

**Comment by jrbenny35** [comment]
> Yep all looks good, thank you, nice optimization. I think we should keep this pattern up and use the api for anything that isn't explicitly testing the nimbus ui. Thanks @jrbenny35 !
> 
> Oh also I just noticed, the commit formatting is slightly off, it's not
> 
> ```
> type(chore)
> ```
> 
> it should be
> 
> ```
> chore(nimbus)
> ```

Sorry about that!

**Pull request merged**